oxidation/libparsec/crates/platform_async/tests/eyeballs.rs

```diff
@@ -43,7 +43,11 @@ async fn test_eyeballs() -> Result<(), ()> {
     assert_eq!(Ok(2), dbg!(eyeballs(targets).await));
     let elapsed = start_time.elapsed();
     log::info!("elapsed time {}ms", elapsed.as_millis());
-    assert!(elapsed < Duration::from_millis(900));
+    // The expected time calculated following these rules:
+    // - The first 2 load will take to much time, so we're adding `TIMEOUT * 2`
+    // - The 3th load will success after 300ms, so we add `300`
+    // - We add some value to take account the possible overhead to switch between async task (here 200ms).
+    assert!(elapsed < Duration::from_millis(250 * 2 + 300 + 200));
```

Comment:
I don't think this is the right way to fix this: `elapsed` is very dependent on how fast the machine is, and what is le the lag in polling the futures (I would be surprise if on Windows or in some specific browsers the lag is very high...)

So what we can test is:
- elapsed took less that <expected duration> (if it took less it means the code is not doing what we think, for instance the sleep is not actually sleeping)
- elapsed took a lot more that <expected duration>, of course it is hard to define "a lot more" (hence this test being flaky !) so I'm not sure what/if we should put here. BTW in Python we use this kind of check to ensure the test doesn't have a deadlock, but this is complicated to do here and the test is simple so I think it is not worth it.
- an order of resolution is the tasks, do you think we could easily capture the logs to everything occurs in the right order ?

Reply:
1. Yes we can do that
2. Hard to define maybe but it should take less than 2.3 sec (The duration of the first load)
3. This would complexify the test

Reply:
> This would complexify the test

I agree, The main reason I was wondering about this caplog thing is because it's very easy to do in Pytest and seems like an interesting tool to have in Rust as well ;-)

> Hard to define maybe but it should take less than 2.3 sec (The duration of the first load)

After more thinking, I think the best thing to do for this test is to check the outcome of each job (so something like we should always have id X in succes, id Y,Z in cancelled and id W not started)

This we could add the also add the "duration expected to be more than x" check and "duration expected to be less than y", but let's say add 1s to y (considering 1s is a huge lag, I'm totally not sure it is the case, but it seems a good approximation...)

Reply:
We could test the output log with something like [logtest](https://docs.rs/logtest/latest/logtest/).
But it seems that what we want is to check the result of each load task.
And we could use a channel for that.

1s is an good approximation for the task that take 2.3 seconds, but is not for the one that take 100ms. Maybe if we say 50% of the load task duration or at least 100ms ?

Reply:
> Maybe if we say 50% of the load task duration or at least 100ms ?

👍